tests/Browser/LoginTest.php

```diff
@@ -0,0 +1,221 @@
+                    ->press('Login')
+                    ->assertFocused('email');
+        });
+    }
+
+    /**
+     * Test email or password wrong input keep email.
+     * Case 1-7
+     */
+    public function testUserWrong()
+    {
+        // Get user
+        $user = User::find(2);
+        $this->browse(function ($browser) use ($user) {
+            $browser->visit(config('contants.path.login'))
+                    ->type('email', $user->email)
+                    ->type('password', 'abc@123')
+                    ->press('Login')
+                    ->assertSee('These credentials do not match our records.')
+                    ->assertInputValue('email', $user->email);
+        });
+    }
+
+    /**
+     * A basic browser test user login.
+     * Case 2-1
+     */
+    public function testUserLogin()
+    {
+        // Get user
+        $user = User::find(2);
+        $this->browse(function ($browser) use ($user) {
+            $browser->visit(config('contants.path.login'))
+                    ->type('email', $user->email)
+                    ->type('password', 'lifull@123')
+                    ->press('Login')
+                    ->assertPathIs('/');
+        });
+    }
+
+    /**
+     * A basic browser test user visit path admin.
+     * Case 2-2
+     */
+    public function testUserVisitAdmin()
+    {
+        $this->browse(function ($browser) {
+            $browser->loginAs(User::find(2))
+                    ->visit(config('contants.path.admin'))
+                    ->assertPathIs('/');
+        });
+    }
+
+    /**
+     * A basic browser test admin login.
+     * Case 2-3
+     */
+    public function testAdminLogin()
+    {
+        // Get Admin
+        $user = User::find(1);
+
+        $this->browse(function ($browser) use ($user) {
+            $browser->visit(config('contants.path.login'))
+                    ->type('email', $user->email)
+                    ->type('password', 'lifull@123')
+                    ->press('Login')
+                    ->assertPathIs(config('contants.path.admin'));
+        });
+    }
+
+    /**
+     * A basic browser test admmin visit path /.
+     * Case 2-4
+     */
+    public function testPathVisitUser()
+    {
+        $this->browse(function ($browser) {
+            $browser->loginAs(User::find(1))
+                    ->visit('/')
+                    ->assertPathIs(config('contants.path.admin'));
+        });
+    }
+
+    /**
+     * A basic browser test user logout.
+     * Case 3-1
+     */
+    public function testUserLogout()
+    {
+        $this->browse(function ($browser) {
+            $browser->loginAs(User::find(2))
+                    ->logout()
+                    ->assertGuest();
+        });
+    }
+
+    /**
+     * Test admin logout.
+     * Case 3-2
+     */
+    public function testAdminLogout()
+    {
+        $this->browse(function ($browser) {
+            $browser->loginAs(User::find(1))
+                    ->logout()
```

Comment:
Test case testing is Logout button . in this code is logout function (not same test)

Reply:
This test is an action that not possible to recreate manually , please change

Reply:
I changed